src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableRequestValidation.cs

```diff
@@ -0,0 +1,86 @@
+            nameof(SystemSecurityCryptographyResources.DoNotDisableRequestValidationMessage),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.DoNotDisableRequestValidationDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: s_Description,
+                helpLinkUri: null,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var validateInputAttributeTypeSymbol = compilationStartAnalysisContext.Compilation.GetTypeByMetadataName(WellKnownTypeNames.SystemWebMvcValidateInputAttribute);
+
+                    if (validateInputAttributeTypeSymbol == null)
+                    {
+                        return;
+                    }
+
+                    compilationStartAnalysisContext.RegisterSymbolAction(
+                        (SymbolAnalysisContext symbolAnalysisContext) =>
+                        {
+                            var methodSymbol = (IMethodSymbol)symbolAnalysisContext.Symbol;
+                            var attr = methodSymbol.GetAttributes().FirstOrDefault(s => s.AttributeClass.Equals(validateInputAttributeTypeSymbol));
+
+                            if (attr == null)
+                            {
+                                return;
+                            }
+
+                            var constructorArguments = attr.ConstructorArguments;
+
+                            if (constructorArguments != null &&
+                                constructorArguments.Length == 1 &&
+                                !constructorArguments[0].IsNull &&
+                                constructorArguments[0].Value.Equals(false))
```

Comment:
Probably want to add a `Kind` check to avoid throwing [this](http://source.roslyn.io/#Microsoft.CodeAnalysis/Symbols/TypedConstant.cs,73) exception for `TypedConstantKind.Array`